server/graphql/types/src/types/store.rs

```diff
@@ -32,13 +32,16 @@ impl StoreNode {
         let loader = ctx.get_loader::<DataLoader<NameByIdLoader>>();
 
         let response_option = loader
-            .load_one(NameByIdLoaderInput::new(&store_id, &self.row().name_id))
+            .load_one(NameByIdLoaderInput::new(
+                &store_id,
+                &self.row().name_link_id,
```

Comment:
We should use name_id here and not the name_link_id, e.g. from the name_link row.  Using the name_link_id could be the wrong id, e.g. using this link id might fail because the link id is not a valid name_id anymore...

The name_link_id should be hidden in the repository. Ideally the name_link_row shouldn't even leave the repo layer, e.g. you could have store domain object that contains only the resolved name_id.

We didn't do this consistently because we wanted to keep the change minimal. However, I think here it can be done?

Reply:
That was gonna be another issue...